js/jquery.mapael.js

```diff
@@ -790,46 +790,81 @@
             var animDuration = (zoomOptions.animDuration !== undefined) ? zoomOptions.animDuration : self.options.map.zoom.animDuration;
             var offsetX = 0;
             var offsetY = 0;
-            var coords = {};
 
-            // Get user defined zoom level
-            if (zoomOptions.level !== undefined) {
-                if (typeof zoomOptions.level === "string") {
-                    // level is a string, either "n", "+n" or "-n"
-                    if ((zoomOptions.level.slice(0, 1) === '+') || (zoomOptions.level.slice(0, 1) === '-')) {
-                        // zoomLevel is relative
-                        newLevel = self.zoomData.zoomLevel + parseInt(zoomOptions.level, 10);
+            if (zoomOptions.area !== undefined) {
+                /* An area is given
+                 * We will define x/y coordinate AND a new zoom level to fill the area
+                 */
+                if (self.areas[zoomOptions.area] === undefined) throw new Error("Unknown area '" + zoomOptions.area + "'");
+                var areaMargin = (zoomOptions.areaMargin !== undefined) ? zoomOptions.areaMargin : 10;
+                var areaBBox = self.areas[zoomOptions.area].mapElem.getBBox();
+                var areaFullWidth = areaBBox.width + 2 * areaMargin;
+                var areaFullHeight = areaBBox.height + 2 * areaMargin;
+                zoomOptions.x = areaBBox.x + areaBBox.width / 2;
+                zoomOptions.y = areaBBox.y + areaBBox.height / 2;
+
+                newLevel = Math.min(Math.floor((self.mapConf.width / areaFullWidth - 1) / self.options.map.zoom.step),
```

Comment:
Shouldn't you take into accout  self.options.map.zoom.minLevel and  self.options.map.zoom.maxLevel for the newLevel computation ?

Reply:
Don't worry about it, I forgot about the final check "Make sure we stay in the zoom level boundaries"

Reply:
Yeah, I moved this check below to avoid repeating the same line :)